typescript/widgets/src/components/DatetimeField.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { ChangeEvent } from 'react';
+
+interface Props {
+  timestamp: number | null;
+  onChange: (t: number | null) => void;
+  name?: string;
+}
+
+export function DatetimeField({ timestamp, onChange, name }: Props) {
```

Comment:
Type the props as extensions to the fields they'll be used on, and then spread the `{...props}` onto the input field. See the svg icon components for an example